### ISSUE TITLE
[Caffe2] Error importing ConvTranspose2d to Caffe2 with ONNX

### ISSUE THREAD

**Issue by santi-pdp** [pytorch/pytorch#10667]
## Issue description

Shape is not the expected one as in PyTorch when importing ONNX description into Caffe2 for `ConvTranspose2d` layers. I made a reduced version of a MobileNet, which comprises the encoder w/ separable strided convs and stuff which all work okay, but when it comes to the decoder there are wrong shapes coming out of the "deconvolution" layers. I relate this to the `padding` and `output_padding` combination, which may get lost in ONNX, just expecting one sort of asymmetric padding (could it be? I know PyTorch deconv `padding` is purely symmetric, so `output_padding` helps enforcing the output shape we want by putting zeros in the bottom-right corner, but I don't know what ONNX does). 

## Code example

An example definition of my deconv layers is:

`self.dec1 = nn.ConvTranspose2d(320, 160, 3, stride=2, output_padding=1,  padding=1)`

such that the output resolution should be the exact double (e.g. `7` -> `14`), but it ends up being `14` in PyTorch and `13` in Caffe2 inference. If I change the definition to avoid the paddings and get exact fits without overlaps in the deconv, things change:

`self.dec1 = nn.ConvTranspose2d(320, 160, 2, stride=2, output_padding=0,  padding=0)`

It gives the right shape, so I have no problem anymore.

## System Info

```
Collecting environment information...
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: 9.1.85

OS: Linux Mint 18.1 Serena
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.11.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 387.26
cuDNN version: Probably one of the following:
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.10
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.6.0.20
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.7
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.7.0.4
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn_static.a
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.5
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.5.1.10
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.6
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.6.0.20
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.7
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.7.0.4
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.7.1.3
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn_static.a

Versions of relevant libraries:
[pip3] numpy (1.14.3)
[pip3] numpydoc (0.7.0)
[pip3] torch (0.5.0a0+0169ac5)
[pip3] torchsample (0.1.3)
[pip3] torchvision (0.2.1)
[conda] cuda80                    1.0                           0    soumith
[conda] cuda90                    1.0                  h6433d27_0    pytorch
[conda] cuda91                    1.0                  h4c16780_0    pytorch
[conda] magma-cuda90              2.3.0                         1    pytorch
[conda] pytorch                   0.4.0           py36_cuda9.1.85_cudnn7.1.2_1  [cuda91]  pytorch
[conda] torch                     0.5.0a0+0169ac5           <pip>
[conda] torchvision               0.2.1                    py36_1    pytorch
```

**Comment by garymm** [comment]
Seems like the issue is with Caffe2 running the model, not torch.onnx.export, so removing `module: onnx` label.
